core/state/statedb.go

```diff
@@ -709,6 +714,30 @@ func (s *StateDB) CommitEtxs() (common.Hash, error) {
 	return root, err
 }
 
+func (s *StateDB) GetLatestEpoch() (uint32, error) {
```

Comment:
You should replace this with blockNumber % epochLength

Reply:
Block number % epoch length will increment for every block because it's the remainder. I think you mean integer division, which is block number / epoch length. I did this